gitlab/v4/objects/environments.py

```diff
@@ -55,3 +57,26 @@ def get(
         self, id: Union[str, int], lazy: bool = False, **kwargs: Any
     ) -> ProjectEnvironment:
         return cast(ProjectEnvironment, super().get(id=id, lazy=lazy, **kwargs))
+
+
+class ProjectProtectedEnvironment(ObjectDeleteMixin, RESTObject):
+    _id_attr = "name"
+    _repr_attr = "name"
+
+
+class ProjectProtectedEnvironmentManager(
+    RetrieveMixin, CreateMixin, UpdateMixin, DeleteMixin, RESTManager
```

Comment:
I'm not sure about `UpdateMixin` - does the API allow `PUT` on existing protected environments? Even protected branches don't support editing, so I'd think that no.

Reply:
it looks like it does not per https://docs.gitlab.com/ee/api/protected_environments.html ?

I understand I should remove the UpdateMixin from the inheritance tree, right ?

Reply:
I think that's best @calve yes, otherwise some people get confused and try to use the `update()` methods. Thanks!